UnitySDK/Assets/ML-Agents/Scripts/Grpc/RpcCommunicator.cs

```diff
@@ -434,6 +442,68 @@ void UpdateSentBrainParameters(UnityRLInitializationOutputProto output)
 
         #endregion
 
+
+        #region Handling side channels
+
+        public void RegisterSideChannel(SideChannel sideChannel)
+        {
+            if (m_SideChannelsDict.ContainsKey(sideChannel.ChannelType()))
+            {
+                throw new UnityAgentsException(string.Format(
+                "A side channel with type index {} is already registered. You cannot register multiple " +
+                "side channels of the same type."));
+            }
+            m_SideChannelsDict.Add(sideChannel.ChannelType(), sideChannel);
+        }
+
+        public static byte[] GetSideChannelMessage(Dictionary<int, SideChannel> sideChannels)
+        {
+            using (var memStream = new MemoryStream())
+            {
+                using (var binaryWriter = new BinaryWriter(memStream))
+                {
+                    foreach (var sideChannel in sideChannels.Values)
```

Comment:
should we have an int or byte for the protocol versioning at the start? If the numbering of the channel types ever changes, it'll be messy.

Reply:
Isn't that taken care of by the communicator API version? I would like to avoid creating a new versioning

Reply:
This seems like it's more volatile than the communicator API version, but I'm OK punting on it for now.